btree/btree.c

```diff
@@ -3443,6 +3447,92 @@ static bool index_is_valid(struct level *lev)
 	return (lev->l_idx >= 0) && (lev->l_idx < node_count(lev->l_node));
 }
 
+/**
+ * btree_open_tick function is used to traverse through different states to
+ * facilitate the working of m0_btree_open().
+ *
+ * @param smop     represents the state machine operation
+ * @return int64_t returns the next state to be executed.
+ */
+int64_t btree_open_tick(struct m0_sm_op *smop)
+{
+	struct m0_btree_op    *bop  = M0_AMB(bop, smop, bo_op);
+	struct td             *tree = NULL;
+	int                    i    = 0;
+	struct m0_btree_oimpl *oi   = bop->bo_i;
+	struct segaddr         curr_addr;
+
+	switch(bop->bo_op.o_sm.sm_state)
+	{
+	case P_INIT:
+		if (!addr_is_aligned(bop->b_data.addr)) {
```

Comment:
This check is okay during create_tree but not needed here anymore.

Reply:
Ok.
I added it because we have few cases in UT for the same. Since, create and open kind of have similar functionality, I think it will be prudent to keep it in both places.

Reply:
You might want to add a check for valid address in the node which could be based on node_is_valid or other things but not alignment of the node.

Reply:
Ok
The issue that will arise is that if we remove this check of bad address for m0_btree_open(), then around 4 cases need to be commented out which were checking the invalid address case in m0_btree_ut_basic_tree_oper(). 
Also node_is_valid() check has not been implemented yet.